Swerve/SwerveModules/SwerveModule.java

```diff
@@ -16,24 +16,46 @@
 import edu.wpi.first.units.Units;
 import edu.wpi.first.units.VelocityUnit;
 import edu.wpi.first.units.measure.Angle;
+import edu.wpi.first.units.measure.Current;
 import edu.wpi.first.units.measure.Dimensionless;
 import edu.wpi.first.units.measure.Distance;
 import edu.wpi.first.units.measure.LinearVelocity;
+import edu.wpi.first.units.measure.AngularVelocity;
 import edu.wpi.first.units.measure.Time;
 import edu.wpi.first.units.measure.Velocity;
 import edu.wpi.first.units.measure.Voltage;
 import edu.wpi.first.wpilibj.Timer;
 import edu.wpi.first.wpilibj.smartdashboard.SmartDashboard;
 import edu.wpi.first.wpilibj.sysid.SysIdRoutineLog;
+import org.littletonrobotics.junction.AutoLog;
+import org.littletonrobotics.junction.Logger;
 
 public abstract class SwerveModule {
     public SwerveModuleConstants swerveModuleConstants;
     protected CANcoder angleEncoder;
     protected Rotation2d lastAngle;
-    
+
+    private final ModuleIOInputsAutoLogged inputs = new ModuleIOInputsAutoLogged();
+
     protected LinearVelocity lastVelo;
     protected Time lastVeloTime;
 
+    @AutoLog
+    public static class ModuleIOInputs {
+
+      public SwerveModuleState swerveState = new SwerveModuleState();
+      public SwerveModulePosition swervePose = new SwerveModulePosition();
+
+      public Distance driveMotorDistance = Units.Meters.zero();
+      public LinearVelocity driveVelocityMetersPerSec = Units.MetersPerSecond.zero();
```

Comment:
having units in the name of the inputs is not necessary since their units are described within their respective unit classes

Reply:
changed